src/main/java/seedu/address/model/event/UniqueEventList.java

```diff
@@ -65,6 +66,23 @@ public void remove(Event toRemove) {
         }
     }
 
+    /**
+     * Replaces the event {@code target} in the list with {@code editedEvent}.
+     * {@code target} must exist in the list.
+     * The event identity of {@code editedPerson} must not be the same as another existing event in the list.
+     */
+    public void setEvent(Event target, Event editedEvent) {
+        requireAllNonNull(target, editedEvent);
+
+        int index = internalList.indexOf(target);
+
+        if (target.equals(editedEvent)) {
+            throw new DuplicateEventException();
+        }
```

Comment:
Sorry this should be `target.isSameEvent(editedEvent)` - they're slightly different.

Reply:
Makes sense. Will do

Reply:
Wait we need to use equals because they need to check the contents of `attendeeList` also. Else when `setEvent` is called in `AssignCommand#execute`, a duplicate is always detected. Check the latest commit from my branch to see the errors from the failed Java CI PR Run

Reply:
Oh ya! I'm so sorry, I was wrong.